patch/patch_crds.sh

```diff
@@ -0,0 +1,78 @@
+#!/bin/bash
+#
+# Copyright (c) 2012-2018 Red Hat, Inc.
+# This program and the accompanying materials are made
+# available under the terms of the Eclipse Public License 2.0
+# which is available at https://www.eclipse.org/legal/epl-2.0/
+#
+# SPDX-License-Identifier: EPL-2.0
+#
+# This script is required to make automatically generated CRDs compatible with
+# Kubernetes >1.18
+#
+# Patching is done in-place. Note that the formatting output by yq is slightly
+# different than the one output by operator-sdk generate commands: list items
+# are indented one level further:
+#     list:            list:
+#     - item1    -->     - item1
+#     - item2            - item2
+#
+
+set -e
+
+COMPONENTS_CRD_PATH="deploy/crds/workspace.che.eclipse.org_components_crd.yaml"
+ROUTINGS_CRD_PATH="deploy/crds/workspace.che.eclipse.org_workspaceroutings_crd.yaml"
+
+# CRD path from root to status field, in jq filter format
+STATUS_PATH='.spec.validation.openAPIV3Schema.properties["status"]'
+# CRD path podAdditions to containerPorts required field, in jq filter format
+# Note this path takes one jq arg: "CONTAINERS_FIELD" can be used to specify
+# "containers" or "initContainers", as both have containerPorts that must be patched.
+PODADDITIONS_PATH='.properties["podAdditions"].properties[$CONTAINERS_FIELD].items.properties["ports"].items.required'
+
+# Full jq path to container ports required field in components CRD.
+COMPONENTS_SPEC_PATH="${STATUS_PATH}"'.properties["componentDescriptions"].items'"${PODADDITIONS_PATH}"
+# Full jq path to container ports required field in workspaceroutings CRD.
+ROUTINGS_SPEC_PATH="${STATUS_PATH}${PODADDITIONS_PATH}"
+
+# Update components CRD using yq; no-op if already patched.
+# Args:
+#  $1 - podAdditions field to update ("containers" or "initContainers")
+function update_components_crd() {
+  field="$1"
+  local yq_check_script="${COMPONENTS_SPEC_PATH}"' | index("protocol")'
+  local yq_patch_script="${COMPONENTS_SPEC_PATH}"' |= . + ["protocol"]'
+  already_patched=$(yq -r "$yq_check_script" "$COMPONENTS_CRD_PATH" --arg "CONTAINERS_FIELD" "$field")
+  if [[ "$already_patched" == "null" ]]; then
+    yq -Y -i "$yq_patch_script" "$COMPONENTS_CRD_PATH" --arg "CONTAINERS_FIELD" "$field"
```

Comment:
Seems we have different yq... it does not work for me =(
![Screenshot_20200508_080946](https://user-images.githubusercontent.com/5887312/81372545-55e19d00-9103-11ea-81a1-d6a1d7efd397.png)

Reply:
Ah you need to update `yq`; in-place editing was added more recently.

Reply:
The precise requirements are

- `jq`: version 1.5.0 or higher (https://github.com/stedolan/jq/releases/tag/jq-1.5)
- `yq`: version 2.9.0 or higher (https://github.com/kislyuk/yq/releases/tag/v2.9.0)

Reply:
OK, thanks for the clarification.
Previously, I installed it with apt-get, but seems new version is not available there. I'll download the latest versions manually.